[9.0] port module delivery_carrier_label_dispatch renamed to delivery_carrier_label_batch

yvaucher opened the pull request:
- change name to batch to match with change of `picking_dispatch` name to
  `stock_batch_picking`
- mode description in README.rst
- move files in models and views directories
- change headers to small license header
- use api 8.0
- fix and improve tests

depends on:
- [x] https://github.com/OCA/stock-logistics-workflow/pull/241

yvaucher committed:
[9.0] port module delivery_carrier_label_dispatch

- change name to batch to match with change of `picking_dispatch` name to
`stock_batch_picking`
- mode description in README.rst
- move files in models and views directories
- change headers to small license header
- use api 8.0
- fix and improve tests

rafaelbn commented:
@carlosdauden you must be interested. :smile:  cc @pedrobaeza

yvaucher committed:
Update translations

yvaucher committed:
Cannot update options due to limitation of api 8.0

mdietrichc2c commented on delivery_carrier_label_batch/wizard/apply_carrier.py:
```diff
@@ -0,0 +1,39 @@
+        domain = self._check_domain(batch_ids)
+        batchs = batch_obj.search(domain)
+        batchs.write({'carrier_id': self.carrier_id.id})
+        batch_obj.action_set_options(batchs)
```
Should be `batchs.action_set_options()` instead.

mdietrichc2c commented on delivery_carrier_label_batch/models/stock_batch_picking.py:
```diff
@@ -0,0 +1,112 @@
+                'carrier_id': rec.carrier_id.id,
+                'option_ids': [(6, 0, rec.option_ids)],
+            }
+            rec.related_picking_ids.write(options_datas)
```
The field `related_picking_ids` does not exist on the `stock.batch.picking` model ; it should be `picking_ids`.

edit: Î compared with the 7.0 code, it must be `picking_ids` to be consistent.

yvaucher commented:
@mdietrichc2c fixed both

mdietrichc2c commented:
@yvaucher :+1:

mdietrichc2c commented on delivery_carrier_label_batch/models/stock_batch_picking.py:
```diff
@@ -0,0 +1,111 @@
+        for rec in self:
+            options_datas = {
+                'carrier_id': rec.carrier_id.id,
+                'option_ids': [(6, 0, rec.option_ids)],
```
Should be `rec.option_ids.ids`

mdietrichc2c committed:
Use IDs instead of recordset for write

yvaucher commented:
Restarted travis as OCA/stock-logistics-workflow#241 was merged

Commit:
rename field tracking_id to package_id

the field has been renamed in the base module

mdietrichc2c committed:
Remove self.pool + add correct attachment name

yvaucher commented:
Thanks @mdietrichc2c for the fix

mdietrichc2c committed:
View changed in batch_stock_picking

Commit:
Fix condition

Commit:
Thread label generation to make it faster

Port of https://github.com/OCA/delivery-carrier/pull/27

mdietrichc2c committed:
Fix onchange for batch picking

mdietrichc2c committed:
Correct error treatment

Commit:
Fix empty pdf issue

Probably due to usage of multiple environment and cache. datas can't be
read from shipping.label but it can be read from ir.attachment

mdietrichc2c committed:
Add pick/pack info to all exceptions

mdietrichc2c committed:
e.message deprecated from 2.6

Commit:
Allow to print labels on operations without product

Commit:
Fix concurrent error on batch labels

When there is more than 1 pack in a picking, 2 or more thread worker
might be in concurrence on the same picking, which makes one of the
transaction fails, resulting in the abort of the whole label generation.

The solution implemented is to change the way of passing the tasks to
the thread workers. Instead of passing a label at a time, we pass a
group of label. A group of label actually contains all the labels to
generate for one picking.

Commit:
Do not regenerate existing labels unless forced + pylint fix

grindtildeath commented on delivery_carrier_label_batch/wizard/generate_labels.py:
```diff
@@ -0,0 +1,245 @@
+import logging
+import openerp
+import threading
+from contextlib import closing, contextmanager
```
```suggestion
from contextlib import contextmanager
```
`closing` must be removed to make travis happy

grindtildeath reviewed (commented):
I guess there were enough changes since the opening of this PR to merge it when travis is green.

rousseldenis commented:
@yvaucher Could you fix travis with @grindtildeath suggestion ?

Commit:
[9.0][ADD] support of zpl labels

sebalix committed:
[IMP] delivery_carrier_label_batch: add support for ZPL labels

i-vyshnevska commented on delivery_carrier_label_batch/wizard/generate_labels.py:
```diff
@@ -0,0 +1,282 @@
+        label_obj = self.env['shipping.label']
+        domain_file_type = ('file_type', '=', f_type)
+        if not f_type:
+            domain_file_type = ('file_type', 'in', self.get_file_types())
```
I thought file_type is mandatory, anyhow if f_type=None doesn't this meant that we should search for all files? do  we need to call get_file_types here?

fyi in v12 file_type was changed from selection to char field, maybe worth to mention in code when will work on migration and writing tests

sebalix replied:
There are existing calls to this method IIRC (outside of this module) that are not aware of the file type, so we need to backward-compatible. :+1: to avoid a call to `get_file_types()`.

rousseldenis commented:
@yvaucher @sebalix Could you remove travis commit as it has been done in a separate PR to keep some formalism ?

sebalix commented:
@rousseldenis done!

sebalix commented:
There is a regression with our last two commits. I'm working on it.

sebalix commented:
Fixed

rousseldenis commented:
@yvaucher @sebalix Is this ready ?

sebalix commented:
@rousseldenis the last development have been validated by our customer, I rebased my last 3 commits: it's ready to be merged now

rousseldenis commented:
/ocabot merge
